assembly/src/todo/context.rs

```diff
@@ -41,40 +38,146 @@ impl ModuleContext {
 
     /// Returns module path for the module which is currently being compiled.
     pub fn module_path(&self) -> &str {
-        &self.module_path
+        &self.modules.last().expect("no modules").module_path
     }
 
-    /// Returns true if this context is used for compiling a kernel module.
+    /// Returns true if this context is used for compiling a kernel.
     pub fn is_kernel(&self) -> bool {
         self.is_kernel
     }
 
-    /// Returns a [Procedure] located at the specified index in the module.
+    /// Returns a [Procedure] located at the specified index in the module which is currently being
+    /// compiled.
     ///
     /// # Error
-    /// Returns an error if there is no compiled procedure at the specified index.
+    /// Returns an error if there is no compiled procedure at the specified index in the current
+    /// module.
     pub fn get_local_proc(&self, index: u16) -> Result<&Procedure, AssemblerError> {
-        self.local_procs
-            .get(index as usize)
-            .ok_or_else(|| AssemblerError::undefined_proc(index))
+        self.modules
+            .last()
+            .expect("no modules")
```

Comment:
Maybe a gentler way is to have a dedicated error variant for the case of an inconsistent modules state? I understand that is unreachable (or should be), but we are returning a result anyway